lib/std/json.zig

```diff
@@ -1708,6 +1708,9 @@ fn parseInternal(comptime T: type, token: Token, tokens: *TokenStream, options:
 }
 
 pub fn parse(comptime T: type, tokens: *TokenStream, options: ParseOptions) !T {
+    if (comptime std.meta.trait.hasFn("jsonParse")(T)) {
```

Comment:
This will only work at the top level: the internal recursion uses `parseInternal`.

Reply:
Whoops, for all examples I tried I somehow never had one where the root did not have `jsonParse` implemented. What's stopping us from simply keeping a boolean on `TokenStream` to know if we need the lookahead or not?

Reply:
I have investigated, and the token parser indeed badly needs an overhaul. I couldn't find a simple enough way to get this properly done without a refactor, but I can't do that right now, so for the time being I will close this PR.